tools/nix/flake.lock

```diff
@@ -10,11 +10,11 @@
         ]
```

Comment:
Can you restore this file to the content that's in `develop`, then run from the repo root:
```shell
nix flake lock --update-input rust-overlay ./tools/nix
```
to selectively update just the dependency we need to support Rust 1.71.1? The rest should be bumped separately and in automated fashion once we land #5526 and fix the scheduled runs.